tracer/test/Datadog.Trace.Tests/Logging/DatadogLoggingTests.cs

```diff
@@ -247,6 +248,54 @@ public void DuringStartup_OldLogFilesGetDeleted()
             ignoredLogFiles.Should().Be(3);
         }
 
+        [Fact]
+        public void WritingToAnInvalidPathDoesntCauseErrors()
+        {
+            string directory;
+
+            if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            {
+                // Windows
+                directory = @"Z:\i-dont-exist";
+            }
+            else
+            {
+                // Linux
+                directory = "/proc/self";
+            }
+
+            // make sure we can't write to it
+            IsDirectoryWritable(directory).Should().BeFalse();
+
+            var config = DatadogLoggingFactory.GetConfiguration(new NameValueConfigurationSource(new()
+            {
+                { ConfigurationKeys.LogDirectory, directory }
+            }));
+            var logger = DatadogLoggingFactory.CreateFromConfiguration(config, DomainMetadata.Instance);
+            logger.Should().NotBeNull();
+
+            logger!.Error("Trying to write an error");
+
+            logger.CloseAndFlush();
+
+            static bool IsDirectoryWritable(string dirPath)
+            {
+                try
+                {
+                    var path = Path.Combine(dirPath, Path.GetRandomFileName());
+                    using (var fs = File.Create(path, bufferSize: 1, FileOptions.DeleteOnClose))
```

Comment:
nit and wondering if the using could be removed and the file disposed in a finally block after the `catch`, to avoid nested try? but not sure it makes a difference 💭

Reply:
Possibly? 😅 It's subtly different then, Note that this is _meant_ to throw - it's only used in this test, not in prod🙂

Reply:
oh of course it's a test my bad it doesnt matter then